src/profiler/profiler.h

```diff
@@ -1098,6 +1102,11 @@ struct ProfileMarker {
   VTUNE_ONLY_CODE(std::unique_ptr<vtune::VTuneInstantMarker> vtune_instant_marker_);
 };
 
+
+class CustomOpProfiler;
+static ProfileDomain custom_op_domain("Custom Operator");
```

Comment:
Can we move custom_op_domain into class CustomOpProfiler?

Reply:
I have removed the unused forward declearation. However, I think we need to keep custom_op_domain here because if we move it into CustomOpProfiler there will be a mutual dependency issue.

Reply:
This design looks rather strange to me. @arcadiaphy 's suggestion seems more reasonable. Why would it cause mutual dependency and why can't we resolve it?

Reply:
because if we move it into CustomOpProfiling then we have to include custom_op_profiler.h in profiler.h, but custom_op_profiler.h is already including profiler.h

Reply:
It used in a similar fashion here: https://github.com/apache/incubator-mxnet/blob/master/src/c_api/c_api_profile.cc#L41

Reply:
We actually only need to use custom_op_domain in profiler.h

Reply:
> because if we move it into CustomOpProfiling then we have to include custom_op_profiler.h in profiler.h, but custom_op_profiler.h is already including profiler.h

Why is it the case? You only move a `ProfileDomain` object into `CustomOpProfiling` class which already contains a data member `static ProfileDomain domain_`. I don't understand why it requires additional header file.

Reply:
But that way, to access custom_op_domain, we need to add "include custom_op_profiler.h" to profiler.h